### PR TITLE
Use Pipelines on HttpContext

### PR DESCRIPTION
**Pull request by benaadams**
Guessing it would be a regression until https://github.com/aspnet/AspNetCore/pull/7110 is merged?

/cc @davidfowl

### CODE REVIEW

**Comment by benaadams** [comment]
Guessing it would be a regression until https://github.com/aspnet/AspNetCore/pull/7110 is merged?

File: src/Benchmarks/Middleware/PlaintextMiddleware.cs [hunk 1]
```diff
@@ -32,6 +32,19 @@ public Task Invoke(HttpContext httpContext)
...
 
+#if NETCOREAPP3_0
+        public async static Task WriteResponse(HttpResponse response)
+        {
+            var payloadLength = _helloWorldPayload.Length;
+            response.StatusCode = 200;
+            response.ContentType = "text/plain";
+            response.ContentLength = payloadLength;
+
+            var pipe = response.BodyPipe;
```
Comment (davidfowl) [inline src/Benchmarks/Middleware/PlaintextMiddleware.cs]: Why not call WriteAsync on PipeWriter?

Comment (benaadams) [reply src/Benchmarks/Middleware/PlaintextMiddleware.cs]: Good idea :)

**Comment by benaadams** [comment]
Not needed

**Pull request closed without merging**